tests/Operation/RenameCollectionFunctionalTest.php

```diff
@@ -0,0 +1,168 @@
+    {
+        (new CommandObserver())->observe(
+            function (): void {
+                $server = $this->getPrimaryServer();
+
+                $insertOne = new InsertOne($this->getDatabaseName(), $this->getCollectionName(), ['x' => 1]);
+                $writeResult = $insertOne->execute($server);
+                $this->assertEquals(1, $writeResult->getInsertedCount());
+
+                $operation = new RenameCollection(
+                    $this->getDatabaseName(),
+                    $this->getCollectionName(),
+                    $this->getDatabaseName(),
+                    $this->toCollectionName,
+                    ['writeConcern' => $this->createDefaultWriteConcern()]
+                );
+
+                $operation->execute($server);
+            },
+            function (array $event): void {
+                $this->assertObjectNotHasAttribute('writeConcern', $event['started']->getCommand());
+            }
+        );
+    }
+
+    public function testRenameExistingCollection(): void
+    {
+        $server = $this->getPrimaryServer();
+
+        $insertOne = new InsertOne($this->getDatabaseName(), $this->getCollectionName(), ['_id' => 1]);
+        $writeResult = $insertOne->execute($server);
+        $this->assertEquals(1, $writeResult->getInsertedCount());
+
+        $operation = new RenameCollection(
+            $this->getDatabaseName(),
+            $this->getCollectionName(),
+            $this->getDatabaseName(),
+            $this->toCollectionName
+        );
+        $commandResult = $operation->execute($server);
+
+        $this->assertCommandSucceeded($commandResult);
+        $this->assertCollectionDoesNotExist($this->getCollectionName());
+        $this->assertCollectionExists($this->toCollectionName);
+
+        $operation = new FindOne($this->getDatabaseName(), $this->toCollectionName, []);
+        $cursor = $operation->execute($server);
+        $this->assertSameDocument(['_id' => 1], $cursor);
+    }
+
+    public function testRenameExistingCollectionExistingTarget(): void
+    {
+        $server = $this->getPrimaryServer();
+
+        $insertOne = new InsertOne($this->getDatabaseName(), $this->getCollectionName(), ['_id' => 1]);
+        $writeResult = $insertOne->execute($server);
+        $this->assertEquals(1, $writeResult->getInsertedCount());
+
+        $insertOne = new InsertOne($this->getDatabaseName(), $this->toCollectionName, ['_id' => 1]);
+        $writeResult = $insertOne->execute($server);
+        $this->assertEquals(1, $writeResult->getInsertedCount());
+
+        $this->expectException(CommandException::class);
+        $this->expectExceptionCode(self::$errorCodeNamespaceExists);
+        $operation = new RenameCollection(
+            $this->getDatabaseName(),
+            $this->getCollectionName(),
+            $this->getDatabaseName(),
+            $this->toCollectionName
+        );
+        $commandResult = $operation->execute($server);
```

Comment:
```suggestion
        $operation->execute($server);
```

Even if the operation doesn't throw for some reason (and the test ultimately fails), this return value is never used. Consider what was done in the `testDefaultWriteConcernIsOmitted` and session tests.